### PR TITLE
Java counting sort

### PR DESCRIPTION
**Pull request by gdhameeja**
This adds Counting Sort implemented in java. 
It returns the sorted array. 
Here is the wiki link : https://en.wikipedia.org/wiki/Counting_sort

### CODE REVIEW

**Comment by sangamcse** [comment]
Comment on d133706867a6f2e808aa19d22c224aabcc297007, file counting_sort/Java/count_sort.java, line 28.

Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.pyjava`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp3fxanglr/counting_sort/Java/count_sort.java
+++ b/tmp/tmp3fxanglr/counting_sort/Java/count_sort.java
@@ -25,7 +25,7 @@
          */
         int maximum_number = getMax(arr);              // get the max number in the array in order to declare count array.
         int[] count_arr = new int[maximum_number + 1]; // maximum_number + 1 as we need to correctly
-                                                      //map each index with the number in the array 
+                                                      //map each index with the number in the array
 
         for (int i = 0; i < count_arr.length; i++) {
             count_arr[i] = 0;
```

**Comment by sangamcse** [comment]
Comment on d133706867a6f2e808aa19d22c224aabcc297007, file counting_sort/Java/count_sort.java, line 40.

Line contains following spacing inconsistencies:
- Tabs used instead of spaces.

*Origin: SpaceConsistencyBear, Section: `all.pyjava`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp3fxanglr/counting_sort/Java/count_sort.java
+++ b/tmp/tmp3fxanglr/counting_sort/Java/count_sort.java
@@ -37,7 +37,7 @@
 
         for (int i = 1; i < count_arr.length; i++) {
             count_arr[i] += count_arr[i - 1]; // rolling sum to get the count of numbers less than the
-            				      // current index in the original array
+                                  // current index in the original array
         }
 
         int[] sorted_arr = new int[arr.length]; // declare and initialize the sorted array so elements from
```

**Comment by gdhameeja** [comment]
I think the patch suggests I have CountSort.java as the filename. I changed that to count_sort.java to adhere to the repo coding guidelines. Can you provide a new patch please?
@sangamcse

**Comment by gdhameeja** [comment]
I converted all tabs to spaces. Please let me know in case of any other issues.

**Comment by sangamcse** [comment]
Comment on 8f1a8860a92858697b395756cf89a6582b1a5b54, file counting_sort/Java/CountSort.java, line 28.

Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.pyjava`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpc_mpjr4g/counting_sort/Java/CountSort.java
+++ b/tmp/tmpc_mpjr4g/counting_sort/Java/CountSort.java
@@ -25,7 +25,7 @@
          */
         int maximum_number = getMax(arr);              // get the max number in the array in order to declare count array.
         int[] count_arr = new int[maximum_number + 1]; // maximum_number + 1 as we need to correctly
-                                                      //map each index with the number in the array 
+                                                      //map each index with the number in the array
 
         for (int i = 0; i < count_arr.length; i++) {
             count_arr[i] = 0;
```

File: counting_sort/Java/count_sort.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import java.util.Arrays;
+import java.util.Collections;
+
+class CountSort {
+    /**
```
Comment (sangamcse) [inline counting_sort/Java/count_sort.java]: Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.pyjava`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpn7_d42aj/counting_sort/Java/count_sort.java
+++ b/tmp/tmpn7_d42aj/counting_sort/Java/count_sort.java
@@ -25,7 +25,7 @@
          */
         int maximum_number = getMax(arr);              // get the max number in the array in order to declare count array.
         int[] count_arr = new int[maximum_number + 1]; // maximum_number + 1 as we need to correctly
-                                                       //map each index with the number in the array 
+                                                       //map each index with the number in the array
 
         for (int i = 0; i < count_arr.length; i++) {
             count_arr[i] = 0;
```

Comment (gdhameeja) [reply counting_sort/Java/count_sort.java]: @sangamcse Hey Sangam, I am unable to apply patches. 
These are the errors I get ->
error: counting_sort/Java/CountSort.java: No such file or directory
error: patch failed: counting_sort/Java/count_sort.java:37
error: counting_sort/Java/count_sort.java: patch does not apply
error: counting_sort/Java/CountSort.java: No such file or directory
Can you please help me with this?

Comment (sangamcse) [inline counting_sort/Java/count_sort.java]: Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.pyjava`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpwep5ebii/counting_sort/Java/count_sort.java
+++ b/tmp/tmpwep5ebii/counting_sort/Java/count_sort.java
@@ -25,7 +25,7 @@
          */
         int maximum_number = getMax(arr);              // get the max number in the array in order to declare count array.
         int[] count_arr = new int[maximum_number + 1]; // maximum_number + 1 as we need to correctly
-                                                        //map each index with the number in the array 
+                                                        //map each index with the number in the array
 
         for (int i = 0; i < count_arr.length; i++) {
             count_arr[i] = 0;
```

**Pull request closed without merging**